schema.graphql

```diff
@@ -78,10 +78,11 @@ type TokenholderDelegation implements Delegation @entity {
   delegators: [Account!] @derivedFrom(field: "delegatee")
 }
 
-type ConfirmedOperator @entity {
+type PREOperator @entity {
+  # ID is the staking provider address
   id: ID!
```

Comment:
I think ID should be the operator address, not the staking provider because we are in the PRE**Operator** entity and we should add a relationship between `PREOperator` and `StakeData` entity.

Reply:
That's a good point. I agree with ID being the operator's address.

But I don't see clearly how to add a relationship between PREOperator and StakeData. Do you mean a [Reverse Lookup](https://thegraph.com/docs/en/developer/create-subgraph-hosted/#reverse-lookups)? This will imply to add a new field in StakeData (probably called preoperator), and in my opinion, we shouldn't mix Threshold Network Staking (StakeData) with an Threshold application of the network (PRE).

Or maybe you mean to use One-To-Many relationship? In that case, do you agree with this schema?

`type PREOperator @entity {
  # ID is the operator address
  id: ID!
  confirmationTimestamp: BigInt
  stakes: [StakeData!]!
}`

The PREOperator entity will include an array of all stakeDatas that use this operator.

What do you think?

Reply:
Yep makes sense. But it should probably be a one-to-one relationship because of this requirement https://github.com/nucypher/nucypher-contracts/blob/main/contracts/contracts/SimplePREApplication.sol#L193 cc @cygnusv.

Another solution could be a schema like this:
```
type StakeData @entity {
  ...
  applications: [Application!] @derivedFrom(field: "stake")
}

interface Application {
  id: ID!
  stake: StakeData!
  authorizedStakeAmount: BigInt!
}

type PREApplication implements Application @entity {
  id: ID! // id = "operatorAddress-preapplication"
  stake: StakeData!
  authorizedStakeAmount: BigInt!
  confirmationTimestamp: BigInt!
 ... // other fields related to the pre app
}

// other applications in the future eg. RandomBeaconApplication
type RandomBeaconApplication implement Application @entity {
...
}
```

So in that case we are ready for multi-staking app authorization and the `applications` field makes sense in the `StakeData` entity because we authorize applications to use a given stake. What do you think? @manumonti @cygnusv ofc we can address it in a separate PR just a thought.

Reply:
> Yeah, that's right and it works for me. I'm curious about David's opinion.

On the other hand, this is probably a rare scenario (but I'm not sure) and there will be 3 apps (at least for now: tbtc, random beacon, and pre) so I don't mind iterating through `applications` from the `StakeData` entity and finding the app which contains `-preapplication` in the ID and then removing it from the store. Just a thought. Both solutions work for me 👍.

Reply:
I finally created a issue addressing this. So if @r-czajkowski agree, we can continue the conversation there.

https://github.com/threshold-network/subgraph/issues/23